src/arrayMethodJoin.js

```diff
@@ -5,8 +5,36 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
```

Comment:
it would be smart to add default separator - logic with actualSeparator wouldn't be needed 😄 
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```